Services/Export/classes/class.ilImport.php

```diff
@@ -292,6 +292,15 @@ protected function doImportObject($dir, $a_type, $a_component = "", $a_tmpdir =
 			$comp = $expfile["component"];
 			$class = ilImportExportFactory::getImporterClass($comp);
 
+			// log a warning for inactive page component plugins, but continue import
+			// page content will be imported, but not its additional data
+			// (other plugins throw an exception in ilImportExportFactory)
+			if ($class == '')
```

Comment:
I would recommend a stricter check using `===`.

Reply:
Why? This would only let other invalid values through to the new $class call, wich would fail, instead of logging the warning.

Reply:
Because an additional case was created in this PR. This case is returning an empty string if the class was not found. 
But if this PR should also handle values like `false`, `nulll`,  `0` or any other possible value that leads to `true`, I would recommend to throw an exception in `ilImportExportFactory::getImporterClass()` and surround this part with an `try-catch`-block instead.

Reply:
The empty string case is handled identically for "==" and "===".

Internally getImporterClass pretty much ensures that it always returns strings, no need to test this and throw exceptions.

ilImport on the other hand could only be sure if getImporterClass would define a return type string, wich is not possible yet. So it trows a warning for identifiable invalid values.

Reply:
The bug handled by this PR was introduced with exportable page component plugins which should not prevent an import if their importer class isn't available. In this case the ilImportExportFactory::getImporterClass() returns an empty class name instead of throwing an exception which would be the case for missing importers of any other component type. So it doen't matter, how iImport checks if the returned class name is empty. 

A stricter variant would be:
if (!class_exists($class))
{
	$this->log->warning("no class found for component: $comp");
	continue;
}

Reply:
I can only give you my suggestions, if the maintainers find this solution is the best for the project, i have to accept this. My opinion on a function like this would be:
- It succeeds: The class name will be returned.
- It fails: An exception will be thrown.

All I can say is that an empty string is more ambigious than an exception.  But as said before this is my vision of an optimal method and must not necessarily corresponds with your optimal vision.

Reply:
"All I can say is that an empty string is more ambigious than an exception."

No doubt about that. But you suggested to replace "==" with "===". Not to introduce an exception.